### DISCUSSION TITLE
Strongly recommend: The official Flet framework should take action soon to create a relatively stable packaging environment for versioning

### DISCUSSION THREAD

**Discussion by LegendaryPistachio** [flet-dev/flet#4297] [Ideas]
The immediate priority is to provide a guideline or checklist for packaging environment dependency versions with the release of new versions. Alternatively, the official team could create a cloud packaging solution for the Flet framework.

The Flet framework inherits the complex packaging characteristics of Flutter, facing numerous dependencies and severe compatibility conflicts among them. The packaging environment lacks stability, resulting in inconsistencies in the packaging experience. For example, there are many dependency versions, and user choices are often arbitrary, leading to bugs during packaging. The success rate of packaging is low, which significantly impacts the success rate of users' projects.

It is strongly recommended that the official team of the Flet framework work towards establishing a relatively stable packaging environment for versioning in the near future. The urgent task is to configure a guideline or checklist for packaging environment dependency versions with the release of new versions. Another option is to create a cloud packaging service for the Flet framework. This will help swiftly address the current problems of high error rates and low success rates in Flet framework packaging, eliminating the negative perception among users. The lack of a relatively stable packaging environment for versioning is critical since it is fundamentally what users criticize and dislike, which hinders the global promotion and adoption of the Flet framework.

Whether the Flet framework can overcome the dilemma of easy development but difficult packaging, and escape the niche language predicament, depends on whether it can make changes in this area. This shouldn't be difficult; it just requires action. In the long term, it should smoothly facilitate the release of a localized packaging tool for Android within the Flet framework.

Finally, I would like to say: The many excellent features of the Flet framework make it stand out in the field of front-end development and spread like wildfire. It can be foreseen that its future will be flourishing, and the prospects are worth looking forward to.

进行回复...

**Comment by ap4499** [comment]
Have you got specific scenarios that you have debugged? What are the issues that you are finding with packing?

**Reply by LegendaryPistachio** [reply]
will take the stable version flet0.24.1 as an example to share my personal experience.First,I used it to complete the code writing,and then I planned to package the code into an Android app on Windows 11.This required configuring and installing dependencies.Secondly,let me talk about this configuration and installation of dependencies.At this point,I was very confusedwhich version of JDK is compatible with flet0.24.1?Which version of Android Studio is compatible?Which version of Android SDK is compatible?Which version of Flutter SDK is compatible?Which version of VS Studio is compatible?Which version of Gradle is compatible?The official flet0.24 documentation does not specify or list these compatible versions.As a result,it was difficult to decide,and I was very frustrated.In the end,I tried to choose the latest versions.However,bugs followed one after another,and fixing them seemed endless.Based on the output logs,I judged that the various dependency versions were incompatible and mismatched with each other.In the end,it wasted time and the packaging failed.

**Reply by ap4499** [reply]
If a Matrix was created in the documentation, which listed requirements, would that solve your issue?

Say, Flet version on one axis, and dependencies/requirements on another? The only one that I have of relevance is the bundled Python interpreter (currently 3.12.7).

I think the packaging has moved a long way, very quickly. In an open-source project, we own the bugs as much as the core devs.

Try out 0.25dev, I am finding it contains a lot of improvements, and really encouragingly, I can see a lot of changes that are directly related to community issues/raised on Github issues. Although there are things to iron out in Flet Build, I haven't noticed anything pervasive - they've all been isolated. I think it is just the nature of a project of this scale and complexity, that is currently still at v0.

On the development blog, I believe one of the roadmapped tasks is to have Flet Build being totally self contained, and resolving the entire environment. See the bottom of here: https://flet.dev/blog/flet-new-packaging-pre-release
